drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -84,18 +85,22 @@ impl Handler {
                 // That includes the protocol identifier.
                 // If it is not an ignored protocol, then send an error.
                 if overrun {
-                    if !rx_buf.is_empty() {
-                        if Protocol::from(rx_buf[0]) != Protocol::Ignore {
+                    if rx_bytes != 0 {
+                        if Protocol::from(rx_buf.as_slice()[0])
+                            != Protocol::Ignore
+                        {
+                            let rx_buf = rx_buf.as_slice();
```

Comment:
This might become clearer as I read other bits of the code, but: what is the length of this slice? In isolation I think I would expect it to be `rx_bytes`, but if that's the case why is `rx_bytes` a separate argument?

Reply:
It's the size of the entire buffer... Along with other comments, I'm considering that `RxMsg` should instead wrap the slice and keep track of the size read rather than owning the array on the stack. That would likely make some of the calling code a bit safer.